### DISCUSSION TITLE
redirect_uri issue when using Keycloak as a reverse proxy and using Spring on a different domain from Keycloak

### DISCUSSION THREAD

**Discussion by mjovanc** [keycloak/keycloak#9984] [Q&A]
I have this setup with Keycloak, Spring Boot and NGINX:

```
api.domain.com
domain.com
```

NGINX has two config files. One **api.domain.com** which points to Spring as a reverse proxy, everything works fine with using the API with Keycloak by using for example Postman/cURL.

I have another one **domain.com** that has a reverse proxy to Keycloak. It works fine going to URL `https://domain.com/auth` and log in etc. But when trying to access content on my Swagger at `https://api.domain.com/v1/docs/index.html` it tries to redirect me to `http://api.domain.com/sso/login` which is two things that are not correct. Firstly the protocol and secondly using the subdomain where Keycloak is not on. 

I have set under **Valid Redirect URIs** `https://domain.com/*`. So when a user tried to access that content on my Swagger, should they not be able to redirect to the Keycloak at `https://domain.com/auth` and then redirect back when authenticated?

This is my NGINX config for the api.domain.com:

```
location / {
        proxy_pass http://private-ip1:8080;
        proxy_set_header    Host               $host;
        proxy_set_header    X-Real-IP          $remote_addr;
        proxy_set_header    X-Forwarded-For    $proxy_protocol_addr;
        proxy_set_header    X-Forwarded-Host   $host;
        proxy_set_header    X-Forwarded-Server $host;
        proxy_set_header    X-Forwarded-Port   $server_port;
        proxy_set_header    X-Forwarded-Proto  $scheme;
    }

```

This is my NGINX config for domain.com:

```
# Keyclaok
    location /auth {
        proxy_pass http://private-ip2:8080;
        proxy_set_header    Host               $host;
        proxy_set_header    X-Real-IP          $remote_addr;
        proxy_set_header    X-Forwarded-For    $proxy_protocol_addr;
        proxy_set_header    X-Forwarded-Host   $host;
        proxy_set_header    X-Forwarded-Server $host;
        proxy_set_header    X-Forwarded-Port   $server_port;
        proxy_set_header    X-Forwarded-Proto  $scheme;
    }

```

I don't seem to understand or find information why this is happening. Do anyone have idea how to fix this?

**Comment by cedriic-dev** [comment]
you got any solution for this?
